script/split_tests.py

```diff
@@ -0,0 +1,197 @@
+
+    @property
+    def total_tests(self) -> int:
+        """Return total tests."""
+        return sum([test.total_tests for test in self.children.values()])
+
+    def __repr__(self):
+        """Return representation."""
+        return f"TestFolder(total={self.total_tests}, children={len(self.children)})"
+
+
+def insert_at_correct_position(
+    test_holder: TestFolder, test_path: str, total_tests: int
+) -> None:
+    """Insert test at correct position."""
+    current_path = test_holder
+    for part in test_path.split("/")[1:]:
+        if part.endswith(".py"):
+            current_path.children[part] = TestFile(test_path, total_tests)
+        else:
+            current_path = current_path.children.setdefault(
+                part, TestFolder(os.path.join(current_path.path, part))
+            )
+
+
+def collect_tests(path: str) -> tuple[TestFolder, TestFile]:
+    """Collect all tests."""
+    result = subprocess.run(
+        ["pytest", "--collect-only", "-qq", "-p", "no:warnings", path],
+        check=False,
+        capture_output=True,
+        text=True,
+    )
+
+    if result.returncode != 0:
+        print("Failed to collect tests:")
+        print(result.stderr)
+        print(result.stdout)
+        sys.exit(1)
+
+    folder = TestFolder(path.split("/")[0])
+    insert_at_correct_position(folder, path, 0)
+    max_tests_in_file = TestFile("", 0)
+
+    for line in result.stdout.splitlines():
+        if not line.strip():
+            continue
+        parts = [x.strip() for x in line.split(":")]
+        if len(parts) != 2:
+            print(f"Unexpected line: {line}")
+            sys.exit(1)
+
+        path = parts[0]
+        total_tests = int(parts[1])
+        max_tests_in_file = max(max_tests_in_file, TestFile(path, total_tests))
+
+        insert_at_correct_position(folder, path, total_tests)
+
+    return (folder, max_tests_in_file)
+
+
+def main() -> None:
+    """Execute script."""
+    parser = argparse.ArgumentParser(description="Split tests into n buckets.")
+
+    def check_greater_0(value: str) -> int:
+        ivalue = int(value)
+        if ivalue <= 0:
+            raise argparse.ArgumentTypeError(
+                f"{value} is an invalid. Must be greater than 0"
+            )
+        return ivalue
+
+    parser.add_argument(
+        "bucket_count",
+        help="Number of buckets to split tests into",
+        type=check_greater_0,
+    )
+
+    arguments = parser.parse_args()
+
+    print("Collecting tests...")
+    (tests, max_tests_in_file) = collect_tests("tests")
+    print(
+        f"Maximum tests in a single file are {max_tests_in_file.total_tests} tests (in {max_tests_in_file.path})"
+    )
+    print(f"Total tests: {tests.total_tests}")
+
+    tests_per_bucket = ceil(tests.total_tests / arguments.bucket_count)
+    print(f"Estimated tests per bucket: {tests_per_bucket}")
+
+    if max_tests_in_file.total_tests > tests_per_bucket:
+        raise ValueError(
```

Comment:
If this is an error, we are not going to run the test suite? What is the plan when this happens